### PR TITLE
Display relevant/irrelevant random prior candidates

### PR DESCRIPTION
**Pull request by terrymyc**
This PR aims to display both relevant and irrelevant random records when adding prior knowledge in Exploration and Simulation modes. Improvements are still needed on the back end to improve efficiency.

image

### CODE REVIEW

File: asreview/webapp/api.py [hunk 1]
```diff
@@ -713,35 +713,103 @@ def api_random_prior_papers(project):  # noqa: F401
     the already labeled items.
     """
 
+    subset = request.args.get("subset", default=False, type=bool)
```
Comment (J535D165) [inline asreview/webapp/api.py:716]: Not clear why this is of type `bool`. I would expect the subset to be similar to history subsets for example.

File: asreview/webapp/api.py [hunk 2]
```diff
@@ -713,35 +713,103 @@ def api_random_prior_papers(project):  # noqa: F401
...
 
-    try:
-        pool_random = np.random.choice(pool, 1, replace=False)[0]
-    except Exception:
-        raise ValueError("Not enough random indices to sample from.")
+    payload = {"result": []}
 
-    try:
-        record = read_data(project.project_path).record(pool_random)
+    if not subset:
```
Comment (J535D165) [inline asreview/webapp/api.py:723]: Always better to have `if subset` over if not.

File: asreview/webapp/api.py (hunk 1 above)
Comment (terrymyc) [reply asreview/webapp/api.py:719]: This is an indicator of project mode. In the simulation and exploration modes, the returned random records need to be extracted from relevant and irrelevant subsets.

**Review by J535D165 — APPROVED** [review]

**Comment by J535D165** [comment]
Nice step forward!

**Pull request merged**